src/compressor/zlib/CompressionPluginZlib.cc

```diff
@@ -14,43 +14,11 @@
 
 
 // -----------------------------------------------------------------------------
+#include "acconfig.h"
 #include "ceph_ver.h"
-#include "arch/probe.h"
-#include "arch/intel.h"
-#include "arch/arm.h"
-#include "compressor/CompressionPlugin.h"
-#include "ZlibCompressor.h"
-#include "common/debug.h"
-
-#define dout_subsys ceph_subsys_mon
-// -----------------------------------------------------------------------------
-
-class CompressionPluginZlib : public CompressionPlugin {
-public:
-  bool has_isal = false;
-
-  explicit CompressionPluginZlib(CephContext *cct) : CompressionPlugin(cct)
-  {}
-
-  virtual int factory(CompressorRef *cs,
-                      std::ostream *ss)
-  {
-    bool isal;
-    if (cct->_conf->compressor_zlib_isal) {
-      ceph_arch_probe();
-      isal = (ceph_arch_intel_pclmul && ceph_arch_intel_sse41);
-    } else {
-      isal = false;
-    }
-    if (compressor == 0 || has_isal != isal) {
-      compressor = CompressorRef(new ZlibCompressor(isal));
-      has_isal = isal;
-    }
-    *cs = compressor;
-    return 0;
-  }
-};
```

Comment:
an alternative would be to split __ceph_plugin_init from CompressionPluginZlib.cc and into a separate file (plugin.cc?). I can then include CompressionPluginZlib.cc in libcephd. Let me know if you'd like me to do that.